src/DefaultBuilder/src/WebHostEnvironment.cs

```diff
@@ -81,6 +78,8 @@ internal void CopyPropertiesTo(IWebHostEnvironment destination)
             destination.ApplicationName = ApplicationName;
             destination.EnvironmentName = EnvironmentName;
 
+            ResolveFileProviders(new Configuration());
```

Comment:
Should we skip this if the FileProviders aren't still the NullFileProvider?

Reply:
I'm thinking setting ContentRootPath/WebRootPath should reset their respective IFileProviders though.

Reply:
Yep -- that makes sense.

I'm thinking we might want to get rid of the `ResolveFileProviders` method entirely and just set the providers for `ContentRootPath`/`WebRootPath` in their respective setters.

Also, should we do this:

```diff
- public IFileProvider ContentRootFileProvider { get; set; }
+ public IFileProvider ContentRootFileProvider { get; private set; }
```

E.g. is there a scenario where a user would want to set the file provider independent of the path?

Reply:
I think we should keep `ResolveFileProviders` and have it call into `ResolveWebRootPath`.

> E.g. is there a scenario where a user would want to set the file provider independent of the path?

I could see there being some sort of custom IFileProvider scenario where you set a real path than then manually set a IFileProvider that has better caching or something. That's what concerns me about resetting. Theoretically, someone could set these properties in the opposite order.